File: MORM.Php/Classes/logger.php
<?php

define ("LOG_PATH", "D:/xampp/apache/logs/");

class Logger
{
    private static $instance;

    public static function Instance()
    {
        if (!isset(self::$instance))
            self::$instance = new Logger();
        return self::$instance;
    } 
    
    //--

    protected function Log($tipo, $method, $message)
    {
        $data = date('d.m.Y h:i:s');
        $log = 
            "<log>\n" .
            "<data>$data<data/>\n" . 
            "<tipo>$tipo</tipo>\n" . 
            "<mensagem>$message</mensagem>\n" . 
            "<metodo>$method</metodo>\n" . 
            "</log>\n\n" ;
        $path = LOG_PATH . date("Y.m.d") . ".audit.xml";
        error_log($log, 3, $path);
    }

    public function Debug($method, $message)
    {
        $this->Log("DEBUG", $method, $message);
    }

    public function Erro($method, $message)
    {
        $this->Log("ERRO", $method, $message);
    }

    public function Info($method, $message)
    {
        $this->Log("INFO", $method, $message);
    }

    public function Warning($method, $message)
    {
        $this->Log("WARNING", $method, $message);
    }    
}
?>

File: MORM.CSharp/src/MORM.Utilidade/Utils/Logger.cs
using System;
using System.Collections.Generic;
using System.Configuration;
using System.IO;

namespace MORM.Utilidade.Utils
{
    //-- galeg - 01/05/2018 16:50:34
    public class Logger
    {
        public static string PrgLog { get; set; } = "logging";
        public static DateTime DatLog { get; set; } = DateTime.Today;

        private static bool _inDebug =
            ConfigurationManager.AppSettings[nameof(_inDebug)] == "true";

        private static string GetDirLog()
        {
            var _listDirLog = new List<string>
            {
                ConfigurationManager.AppSettings["dirlog"],
                Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "Log"),
                "c:\\temp\\",
            };

            foreach (var dirLog in _listDirLog)
                if (!string.IsNullOrWhiteSpace(dirLog) && Directory.Exists(dirLog))
                    return dirLog;

            return "c:\\";
        }

        private static string _arqLog;
        public static string ArqLog
        {
            get
            {
                if (string.IsNullOrWhiteSpace(_arqLog))
                    _arqLog = Path.Combine(GetDirLog(), PrgLog + "." + DateTime.Today.ToString("yyyy.MM.dd") + ".xml");
                return _arqLog;
            }

            set
            {
                _arqLog = value;
            }
        }

        protected enum TipoLog
        {
            Debug,
            Info,
            Erro
        }

        protected static void Log(TipoLog tipo, string metodo, string message)
        {
            // gerar novo arquivo quando virada do dia
            if (DatLog != DateTime.Today)
            {
                _arqLog = null;
                DatLog = DateTime.Today;
            }

            var conteudo =
                "<log>" + "\r\n" +
                "<data>" + DateTime.Now.ToString() + "</data>" + "\r\n" +
                "<tipo>" + tipo.ToString() + "</tipo>" + "\r\n" +
                "<metodo>" + metodo + "</metodo>" + "\r\n" +
                "<message>" + message + "</message>" + "\r\n" +
                "</log>" + "\r\n";

            StreamWriter vWriter = new StreamWriter(ArqLog, true);
            vWriter.WriteLine(conteudo);
            vWriter.Flush();
            vWriter.Close();
        }

        public static void Debug(string metodo, string message)
        {
            if (_inDebug)
                Log(TipoLog.Debug, metodo, message);
        }

        public static void Erro(string metodo, string message)
        {
            Log(TipoLog.Erro, metodo, message);
        }

        public static void Info(string metodo, string message)
        {
            Log(TipoLog.Info, metodo, message);
        }
    }
}